runtime/src/main/java/com/dylibso/chicory/runtime/StackFrame.java

```diff
@@ -28,48 +26,30 @@ public class StackFrame {
     public int numberOfValuesToReturn = 0;
     public Value branchConditionValue = null;
 
-    public StackFrame(Instance instance, int funcId, int pc, Value[] args, List<Value> initLocals) {
+    public StackFrame(Instance instance, int funcId, int pc, Value[] args, ValueType[] localTypes) {
         this.instance = instance;
         this.funcId = funcId;
         this.pc = pc;
-        this.locals = new HashMap<>();
-        // pre-initialize everything to 0
-        for (var i = 0; i < initLocals.size(); i++) {
-            var l = initLocals.get(i);
-            var type = l.type() == null ? ValueType.I32 : l.type();
-            // TODO need a cleaner way to initialize?
-            // there are footguns to using the raw Value constructor
-            switch (type) {
-                case I32:
-                    this.setLocal(i, Value.i32(0));
-                    break;
-                case F32:
-                    this.setLocal(i, Value.f32(0));
-                    break;
-                case I64:
-                    this.setLocal(i, Value.i64(0));
-                    break;
-                case F64:
-                    this.setLocal(i, Value.f64(0));
-                    break;
+        this.locals = new Value[args.length + localTypes.length];
+
+        // initialize all locals.
+        for (var i = 0; i < args.length; i++) locals[i] = args[i];
+        for (var i = 0; i < localTypes.length; i++) {
+            ValueType type = localTypes[i];
+            // TODO: How do we initialize non-numeric V128
+            if (type != ValueType.V128) {
+                locals[i + args.length] = Value.zero(type);
```

Comment:
Yeah I wasn't sure on right thing here.  I want it to be obvious it is an issue and had thought this would be seen more readily.  If someone ever removed it I put the throw in there so it would not get lost.

Reply:
@dmlloyd yeah I think I will let this stand for now and see if others have a suggestion.  The best solution would be to figure out how to init a V128.